### PR TITLE
Refactor LgtmDismissAlert function

### PR DESCRIPTION
**Pull request by aibaars**
Rename LgtmDismissAlert to LgtmTransitionNotificationFunction and improve the UI for configuring the transition.

@nickfyson

### CODE REVIEW

File: src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java [hunk 1]
```diff
@@ -121,7 +117,7 @@ private Transition getTransitionParam(Map<?, ?> args) throws InvalidInputExcepti
       throw new InvalidInputException("Missing value for 'transition' parameter");
```
Comment (nickfyson) [inline src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: I'm hitting this error when trying to suppress alerts through the Jira UI.

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: (Haven't yet understood this PR sufficiently to work out why...)

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: Works fine if I go to the function config page for each of the transitions and set the transition type manually. So I guess it's just not getting initialised properly for some reason? (This was with a fresh Jira instance, no migration from a previous plugin version...)

Comment (aibaars) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: I think the workflow.xml needed fixing. Could you try again?

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: Still no luck, I'm afraid. Same error as before, and again goes away if I update the transition type for the post function manually.

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: Also, the post function for the  `unsuppress` transition is being initialised with a transition of type of `suppress`, so I think they're both just on 'default' somehow.

Comment (nickfyson) [reply src/main/java/com/semmle/jira/addon/workflow/LgtmTransitionNotificationFunction.java]: Oh now wait, I'm talking rubbish, that _has_ fixed it!

**Pull request merged**